x-pack/plugins/enterprise_search/server/plugin.ts

```diff
@@ -18,6 +18,7 @@ import {
 import { CustomIntegrationsPluginSetup } from '@kbn/custom-integrations-plugin/server';
 import { PluginSetupContract as FeaturesPluginSetup } from '@kbn/features-plugin/server';
 import { InfraPluginSetup } from '@kbn/infra-plugin/server';
+import type { MlPluginSetup } from '@kbn/ml-plugin/server';
```

Comment:
Why do we need `type` here?

Reply:
Looking at the usages down below, we only need the type rather than the all whole module itself. I think it should be probably the same for the others too. Either way it should work.